testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -1115,14 +1116,29 @@ fn forked_tenure_testing(
         .nakamoto_test_skip_commit_op
         .set(false);
 
+    let mut tenure_b_deadline = None;
+
+    // have the current miner thread just skip block-processing, so tenure C can start
+    relay::fault_injection::skip_stacks_announce();
+    relay::fault_injection::unblock_stacks_announce();
+
     next_block_and(
         &mut signer_test.running_nodes.btc_regtest_controller,
         60,
         || {
             let commits_count = commits_submitted.load(Ordering::SeqCst);
             if commits_count > commits_before {
-                // now allow block B to process if it hasn't already.
-                TEST_BLOCK_ANNOUNCE_STALL.lock().unwrap().replace(false);
+                // once the commit happens, give a grace period for tenure C to be mined and
+                // procesed
+                if let Some(dl) = tenure_b_deadline.as_mut() {
+                    if *dl < get_epoch_time_ms() {
+                        // unblock miner thread, but don't process tenure B just yet
+                        coord_channel.lock().unwrap().announce_new_stacks_block();
+                        *dl = get_epoch_time_ms() + 100_000_000_000;
+                    }
+                } else {
+                    tenure_b_deadline = Some(get_epoch_time_ms() + 10_000);
+                }
```

Comment:
Can you use `Instant`, `Duration`, and a `unblocked_miner` flag here instead? I think it makes a little more clear what the grace period is (and what the `+ 100_000_000_000` line is trying to achieve).

```suggestion
                // once the commit happens, give a grace period for tenure C to be mined and
                // procesed
                if let Some(commit_happened_at) = commit_happened_at.take() {
                    if commit_happened_at.elapsed() > Duration::from_secs(10) && !unblocked_miner {
                        // unblock miner thread, but don't process tenure B just yet
                        coord_channel.lock().unwrap().announce_new_stacks_block();
                        unblocked_miner = true;
                    }
                } else {
                    commit_happened_at = Some(Instant::new());
                    unblocked_miner = false;
                }
```

Reply:
To add to what Aaron said, always prefer `Instant` for local timers because it uses `CLOCK_MONOTONIC` rather than `CLOCK_REALTIME` and will not be affected by time changes from NTP or elsewhere